### DISCUSSION TITLE
Query placeholder for column names in insert statement

### DISCUSSION THREAD

**Discussion by soysandeep** [sidorares/node-mysql2#1628] [Q&A] [answered]
Hi team,

I want to create a common method to "insert" data into any table. 

I want something like this 👇🏻 
```
// columns = ['col1', 'col2']
// values = ['foo', 'bar']
connection.query(`INSERT INTO ${table} (?) VALUES (?)`, [columns, values])
```

I tried running this statement, and it resolves in 👇🏻 
```
INSERT INTO table1 ('col1', 'col2') VALUES ('foo', 'bar');
```
which is incorrect syntax. Because column names should not be wrapped in quotes. If I remove quotes from them, it works fine. Has anyone else faced this problem? Need little help.

**Comment by dougwilson** [comment, marked as answer]
```js
connection.query(`INSERT INTO ${table} (??) VALUES (?)`, [columns, values])
```
